src/mediapipe_internal/mediapipegraphconfig.hpp

```diff
@@ -167,14 +200,33 @@ class MediapipeGraphConfig {
     Status parseNode(const rapidjson::Value& v) {
         try {
             this->setGraphName(v["name"].GetString());
-            this->setGraphPath(v["graph_path"].GetString());
+            if (v.HasMember("base_path")) {
+                std::string providedBasePath(v["base_path"].GetString());
+                if (providedBasePath.back() == '/')
+                    this->setBasePath(providedBasePath);
+                else
+                    this->setBasePath(providedBasePath + "/");
+            } else {
+                if (!getRootDirectoryPath().empty()) {
+                    this->setBasePath(getRootDirectoryPath());
+                    SPDLOG_DEBUG("base_path not defined in config so it will be set to default based on main config directory: {}", this->getBasePath());
+                } else {
+                    SPDLOG_ERROR("Mediapipe {} root directory path is not set.", getGraphName());
+                    return StatusCode::INTERNAL_ERROR;
+                }
+            }
+            if (v.HasMember("graph_path")) {
+                this->setGraphPath(v["graph_path"].GetString());
```

Comment:
If we have relative graph_path eg. "graph_path": "someName.pbtxt" it will search now not in base path dir but in config.json dir?